src/test/scala/scala/tools/refactoring/tests/sourcegen/CommentsUtilsTest.scala

```diff
@@ -0,0 +1,145 @@
+package scala.tools.refactoring.tests.sourcegen
+
+import scala.tools.refactoring.sourcegen.CommentsUtils
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class CommentsUtilsTest {
+  @Test
+  def testStripCommentWithExampleFromTicket1002166() {
+    val source = """
+      object Class {
+        ("//")
+        ("//")
+      }
+    """
+
+   assertEquals(source, CommentsUtils.stripComment(source))
+  }
+
+  @Test
+  def testStripCommentWithMinimalExampleFromTicket1002166() {
+    val source = """("\\")"""
+    assertEquals(source, CommentsUtils.stripComment(source))
+  }
+
+  @Test
+  def testSplitCommentWithSimpleExamples() {
+    testSplitComment("", "", "")
+    testSplitComment("//", "  ",  "//")
+    testSplitComment("""/**/""", "    ", "/**/")
+    testSplitComment("a = 3//test", "a = 3      ", "     //test")
+    testSplitComment("/**/a/*b*/c/*d*/e//f", "    a     c     e   ", "/**/ /*b*/ /*d*/ //f")
+  }
+
+  @Test
+  def testSplitCommentWithMultilineExamples() {
+    val tripleQuote = "\"\"\""
+    
+    /*
+     * Attention:
+     *  These test cases contain a lot of relevant white space chars; make sure that they are not stripped
+     *  away by safe actions.
```

Comment:
This is not scalable. Please add a `$` sign (or any other sign) at the end of such lines and strip it away later.